torch/distributed/algorithms/quantization.py

```diff
@@ -0,0 +1,130 @@
+import functools
+import torch
+import torch.autograd
+import torch.distributed as dist
+import torch.nn
+
+
+from enum import Enum
+
+
+TORCH_HALF_MIN = torch.finfo(torch.float16).min
+TORCH_HALF_MAX = torch.finfo(torch.float16).max
+
+class QuantType(Enum):
+    FP16 = "fp16"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+def _fp32_to_fp16_with_clamp(tensor: torch.Tensor) -> torch.Tensor:
+    return torch.clamp(tensor, TORCH_HALF_MIN, TORCH_HALF_MAX).half()
+
+def _quantize_tensor(tensor, qtype):
+    if not isinstance(tensor, torch.Tensor):
+        raise RuntimeError(
+            "Invalid function argument. Expected tensor "
+            "to be a single tensor list  of type torch.Tensor."
+        )
+    if (qtype == QuantType.FP16):
+        return _fp32_to_fp16_with_clamp(tensor)
+    else:
+        raise RuntimeError(
+            'Quantization type {} is not supported'.format(qtype)
+        )
+
+def _quantize_tensor_list(tensor_list, qtype):
+    if not isinstance(tensor_list, list) or not all(
+        isinstance(p, torch.Tensor) for p in tensor_list
+    ):
+        raise RuntimeError(
+            "Invalid function argument. Expected "
+            "to be of type List[torch.Tensor]."
+        )
+    if (qtype == QuantType.FP16):
+        input_tensor = [_quantize_tensor(t, qtype) for t in tensor_list]
+        return input_tensor
+    else:
+        raise RuntimeError(
+            'Quantization type {} is not supported'.format(qtype)
+        )
+
+def _dequantize_tensor(tensor, qtype, quant_loss=None):
+    if not isinstance(tensor, torch.Tensor):
+        raise RuntimeError(
+            "Invalid function argument. Expected "
+            "to be a single tensor list  of type torch.Tensor."
```

Comment:
what's this error message mean? should be sth like f"_dequantize_tensor expecting torch.Tensor as input but found {type(tensor)}"